agimus_controller/agimus_controller/warm_start_reference.py

```diff
@@ -0,0 +1,71 @@
+import numpy as np
+import numpy.typing as npt
+
+import pinocchio as pin
+
+from agimus_controller.warm_start_base import WarmStartBase
+from agimus_controller.trajectory import TrajectoryPoint
+
+
+class WarmStartReference(WarmStartBase):
+    def __init__(self) -> None:
+        super().__init__()
+        self._rmodel: pin.Model | None = None
+        self._rdata: pin.Data | None = None
+
+    def setup(self, rmodel: pin.Model) -> None:
+        self._rmodel = rmodel
+        self._rdata = self._rmodel.createData()
+
+    def generate(
+        self,
+        initial_state: TrajectoryPoint,
+        reference_trajectory: list[TrajectoryPoint],
+    ) -> tuple[
+        npt.NDArray[np.float64],
+        list[npt.NDArray[np.float64]],
+        list[npt.NDArray[np.float64]],
+    ]:
+        """
+        Generate initial values for a warm-start of the optimization problem.
+
+        This method uses the provided initial state and reference trajectory to compute:
+        - `x0`: The initial state vector `[q, v]`, where:
+            - `q` is the robot's joint configuration.
+            - `v` is the robot's joint velocity.
+        - `init_xs`: A list of state vectors `[q, v]` constructed from the reference trajectory.
+        - `init_us`: A list of control inputs computed using inverse dynamics (RNEA)
+            based on the reference trajectory.
+
+        Assumes:
+            - The state `x` is represented as `[q, v]`.
+            - `reference_trajectory` contains `TrajectoryPoint` objects with:
+                - `robot_configuration` (`q`).
+                - `robot_velocity` (`v`).
```

Comment:
There is no keyword `Assumes` in google docstrings. This should be part of the description. Also this is basically what is written above...